src/main/java/com/forgeessentials/chat/irc/IrcHandler.java

```diff
@@ -0,0 +1,444 @@
+
+    /* ------------------------------------------------------------ */
+
+    public IrcHandler()
+    {
+        ForgeEssentials.getConfigManager().registerLoader(ModuleChat.CONFIG_FILE, this);
+
+        registerCommand(new CommandHelp());
+        registerCommand(new CommandListPlayers());
+        registerCommand(new CommandMessage());
+        registerCommand(new CommandReply());
+    }
+
+    public static IrcHandler getInstance()
+    {
+        return ModuleChat.instance.ircHandler;
+    }
+
+    public void registerCommand(IrcCommand command)
+    {
+        for (String commandName : command.getCommandNames())
+            if (commands.put(commandName, command) != null)
+                OutputHandler.felog.warning(String.format("IRC command name %s used twice!", commandName));
+    }
+
+    public void connect()
+    {
+        if (bot != null)
+            disconnect();
+
+        OutputHandler.felog.info("Initializing IRC connection");
+        bot = new PircBotX();
+        bot.getListenerManager().addListener(this);
+        bot.setName(botName);
+        bot.setLogin(botName);
+        bot.setVerbose(false);
+        bot.setAutoNickChange(true);
+        bot.setMessageDelay(0);
+
+        bot.setCapEnabled(!twitchMode);
+        if (twitchMode)
+            // Prevent pesky messages from jtv because we are sending too fast
+            bot.setMessageDelay(3000);
+
+        try
+        {
+            OutputHandler.felog.info(String.format("Attempting to join IRC server %s on port %d", server, port));
+            bot.connect(server, port, serverPassword.isEmpty() ? null : serverPassword);
+            bot.identify(nickPassword);
+
+            OutputHandler.felog.info("Attempting to join channels...");
+            for (String channel : channels)
+            {
+                OutputHandler.felog.info(String.format("Attempting to join #%s", channel));
+                bot.joinChannel(channel);
+            }
+            OutputHandler.felog.info("IRC bot connected");
+        }
+        catch (NickAlreadyInUseException e)
+        {
+            OutputHandler.felog.warning("[IRC] Connection failed, assigned nick already in use");
+        }
+        catch (IOException e)
+        {
+            OutputHandler.felog.warning("[IRC] Connection failed, could not reach the server");
+        }
+        catch (IrcException e)
+        {
+            OutputHandler.felog.warning("[IRC] Connection failed: " + e.getMessage());
+        }
+    }
+
+    public void disconnect()
+    {
+        if (bot != null)
+        {
+            bot.shutdown();
+            ircUserCache.clear();
+            bot = null;
+        }
+    }
+
+    public boolean isConnected()
+    {
+        return bot != null && bot.isConnected();
+    }
+
+    /* ------------------------------------------------------------ */
+
+    @Override
+    public void load(Configuration config, boolean isReload)
+    {
```

Comment:
Bot commands should be configurable. Some servers may want their IRC bot to only relay messages,